cmake/modules/DispatchCompilerWarnings.cmake

```diff
@@ -1,6 +1,8 @@
 
-if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")
+if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC" OR MINGW)
   # TODO: someone needs to provide the msvc equivalent warning flags
```

Comment:
I believe this project compiles on Windows using clang-cl.  In that case, all of the compiler options defined below can be passed to the clang-cl compiler.  That surfaces a bunch of warnings, which could be fixed in a follow-up PR.

Reply:
IIRC, `-fms-extensions` are non-conforming extensions and we should not be enabling them. If MinGW requires that, then it should be done under a `MINGW` specific clause.

Reply:
You need `-fms-extensions` because [`__popcnt64` is used in hw_config.h](https://github.com/apple/swift-corelibs-libdispatch/blob/8adb6445f4fbbd7b9db2fa92f2cff90dc582d6c9/src/shims/hw_config.h#L157).  That's a [Microsoft-specific extension](https://learn.microsoft.com/en-us/cpp/intrinsics/popcnt16-popcnt-popcnt64?view=msvc-170).

The source code, as it stands, requires this Microsoft-specific extension when targeting the Windows platform.  I assume you don't need to make it explicit on Windows because you're using `clang-cl` (which enables `-fms-extensions` by default), or the MSVC compiler.

I think the proper condition is "when targeting Windows but not using a MSVC-compatible compiler`, let me try to refine that if clause.